### ISSUE TITLE
Getting an IllegalStateException were only IOExceptions are caught

### ISSUE THREAD

**Issue by zyclonite** [pengrad/java-telegram-bot-api#45]
Getting an IllegalStateException but only IOException will result in a failed async response

```
Exception in thread "BotHttp Dispatcher" com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220)
	at com.google.gson.Gson.fromJson(Gson.java:879)
	at com.google.gson.Gson.fromJson(Gson.java:844)
	at com.google.gson.Gson.fromJson(Gson.java:793)
	at com.pengrad.telegrambot.impl.TelegramBotClient$1.onResponse(TelegramBotClient.java:34)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:133)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:388)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:209)
	... 9 more
```

**Comment by pengrad** [comment]
How to reproduce or debug? What is request/response?

**Comment by zyclonite** [comment]
this only happens once in a while ... for me it looks like there is a malformed response from the telegram endpoint that could not be parsed

simply catching the exception would already help a lot i guess
```
    public <T extends BaseRequest, R extends BaseResponse> void send(final T request, final Callback<T, R> callback) {
        client.newCall(createRequest(request)).enqueue(new okhttp3.Callback() {
            @Override
            public void onResponse(Call call, Response response) {
                try {
                    R result = gson.fromJson(response.body().string(), request.getResponseType());
                    callback.onResponse(request, result);
                } catch (IOException | IllegalStateException e) {
                    callback.onFailure(request, e);
                }
            }
```

**Comment by pengrad** [comment]
Ok, thanks

**Comment by pengrad** [comment]
Fixed in 2.1.2
